### ISSUE TITLE
Issue downloading dependencies causing failed builds

### ISSUE THREAD

**Issue by jmc-crash** [microsoft/MixedReality-SpectatorView#251]
Running the SetupRepository.bat file throws exceptions downloading (at least) QR Code dependencies. I do not require these as I wish to use ASA, but the Unity project points at them and therefore requires them for a successful build. Command line output below:

==================================
C:\spectator-view\sv>tools\Scripts\SetupRepository.bat

Enabling symbolic links for the repository.
Configuring the repository to use crlf line endings.
Populating QR Code Dependencies
Exception calling "DownloadFile" with "2" argument(s): "The request was aborted: Could not create SSL/TLS secure
channel."
At C:\spectator-view\sv\tools\Scripts\ExternalDependencyHelpers.ps1:9 char:5
       +    $wc.DownloadFile($url, $zipFile)
       +    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : WebException

Expand-Archive : The path
'C:\spectator-view\sv\tools\Scripts\..\..\external\MixedReality-QRCodePlugin\qrcodeplugin.zip' either does not exist
or is not a valid file system path.
At C:\spectator-view\sv\tools\Scripts\ExternalDependencyHelpers.ps1:10 char:5
    + Expand-Archive -Path $zipFile -DestinationPath "$PSScriptRoot\..\ ...
    + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (C:\spectator-vi...rcodeplugin.zip:String) [Expand-Archive], InvalidOpe
   rationException
    + FullyQualifiedErrorId : ArchiveCmdletPathNotFound,Expand-Archive

Updating spectator view related submodules.
Fixing all symbolic links.

Setup Completed. Press any key to continue...

==================================

**Comment by chrisfromwork** [comment]
This is the second report we've heard around this download failing. I will clone a new repo and take a look.

In hopes of unblocking your work, are you trying to build your own project? If you remove the `QRCODESTRACKER_BINARY_AVAILABLE` from the WSA player settings, you should be able to build in Unity without the QR Code binaries.

**Comment by chrisfromwork** [comment]
I am unable to repro this locally. I did read that this can sometimes be called by the protocol chosen. I have updated a script here that can be tried:
https://github.com/microsoft/MixedReality-SpectatorView/tree/fixDownloadScript
https://github.com/microsoft/MixedReality-SpectatorView/commit/5801702030dbf05aaa3817562fb888991f48ba78

If this commit fixes your issue, we will look to merge it into master. If it doesn't make a difference with your setup, I will delete the branch.

**Comment by jmc-crash** [comment]
I freshly cloned the project and replaced the existing download script with your revised one and the download (and the whole SetupRepository.bat script) works now so that can be merged!

Unfortunately this hasn't corrected my other Unity specific errors which are not actually related to QRCode tracking, but rather ASA. I am just trying to run the "Build2019Demo.Unity" project within the samples directory and am getting the following errors. I can open a new issue for these though as this problem is solved! Thanks!

![Build2019_UnityErrors](https://user-images.githubusercontent.com/54383366/67322778-088a4d80-f56e-11e9-9d75-2633136d64b5.png)

**Comment by chrisfromwork** [comment]
It looks like the symbolic links were setup for half of the dependencies correctly. Since MixedRealityToolkit* isn't showing as a folder, something is wrong.

You may be able to close Unity and rerun SetupRepository.bat to fix things. You might need to delete the Unity project's temp and library directories in addition to running SetupRepository.bat again.

**Comment by jmc-crash** [comment]
Yeah exactly. I tried following your suggestions but unfortunately the errors are still present. Although I tried it twice, just in case.., and I got a set of errors before the usual errors (see image). Perhaps a clue?

![Build2019_UnityErrors_AfterSetupBat](https://user-images.githubusercontent.com/54383366/67339445-c3731500-f587-11e9-9346-3e03f3efc76f.png)

**Comment by chrisfromwork** [comment]
if you open file explorer, do the symbolic directories declared in your Assets folder resolve correctly? From these errors, its hard to tell whether or not the symbolic link is broken or if Unity isn't resolving a working symbolic link correctly.

**Comment by jmc-crash** [comment]
They appear to be resolving correctly through the explorer.

![LinksResolving](https://user-images.githubusercontent.com/54383366/67341231-b6f0bb80-f58b-11e9-801a-ba19892292a2.png)

**Comment by andreiborodin** [comment]
It looks like it isn't actually, as the icon is not that of a folder. Can you open the file in a text editor and copy the contents here?

And try "Open folder location"

**Comment by jmc-crash** [comment]
Open file location takes me here: C:\spectator-view\sv\external\Azure-Spatial-Anchors-Samples\Unity\Assets

And actually opening the AzureSpatialAnchorsPlugin .symlink file opens all of this in the text editor..

![ASAPluginTextEditor](https://user-images.githubusercontent.com/54383366/67342004-a5a8ae80-f58d-11e9-8526-3601c3a638fb.png)

**Comment by andreiborodin** [comment]
Strange, could we jump onto slack to chat and figure out what's going on?

**Comment by jmc-crash** [comment]
Yes, sounds good! I'm not in a place where I can speak at the moment unfortunately, but I can certainly text chat, if that's okay? Otherwise we could arrange a better time?

**Comment by andreiborodin** [comment]
Yes, that works. Please join HoloDevelope here: aka.ms/holodevelopers

And then, join this channel: https://holodevelopers.slack.com/messages/C69D5S5K9

**Comment by andreiborodin** [comment]
The issue was that the symbolic links were not being regenerated for some reason. Manual deletion and "git reset --hard" on the git repo fixes them. I will look into the why SetupRepository or FixSymbolicLinks didn't work.

**Comment by andreiborodin** [comment]
I have the PR to fix this here: #255

**Comment by andreiborodin** [comment]
PR went in, closing the issue. This will be avaialbe in the next release.
